zrml/court/src/lib.rs

```diff
@@ -310,54 +343,16 @@ mod pallet {
             INITIAL_JURORS_NUM.saturating_add(SUBSEQUENT_JURORS_FACTOR.saturating_mul(len))
         }
 
-        // Handy iterator 👍
-        //
-        // Used by `slash_losers_to_award_winners`. Tardy jurors include previously removed
-        // accounts, winners of the losing side, accounts that didn't vote and accounts that voted
-        // after the maximum allowed block, therefore, this function only yields valid winners
-        // and valid jurors that voted on the third or worse outcome.
-        fn non_tardy_jurors_with_votes<'a, 'b>(
-            requested_jurors: &'a mut [(T::AccountId, Juror, T::BlockNumber)],
-            votes: &'b [(T::AccountId, (T::BlockNumber, OutcomeReport))],
-        ) -> impl Iterator<
-            Item = (
-                &'a mut T::AccountId,
-                &'a mut Juror,
-                &'a mut T::BlockNumber,
-                &'b (T::BlockNumber, OutcomeReport),
-            ),
-        > {
-            requested_jurors.iter_mut().filter_map(move |(jai, j, max_block)| {
-                if matches!(j.status, JurorStatus::Tardy) {
-                    return None;
-                }
-                let vote_opt = votes.iter().find(|el| &el.0 == jai).map(|el| &el.1);
-                Some((jai, j, max_block, vote_opt?))
-            })
-        }
-
-        // Useful when dealing with both structures at the same time
-        fn set_stored_and_in_memory_juror_as_tardy(
-            account_id: &T::AccountId,
-            juror: &mut Juror,
-        ) -> DispatchResult {
-            juror.status = JurorStatus::Tardy;
-            Self::set_stored_juror_as_tardy(account_id)
-        }
-
         // Every juror that not voted on the first or second most voted outcome are slashed.
         fn slash_losers_to_award_winners(
-            requested_jurors: &mut [(T::AccountId, Juror, T::BlockNumber)],
-            votes: &[(T::AccountId, (T::BlockNumber, OutcomeReport))],
+            valid_winners_and_losers: &[(&T::AccountId, &OutcomeReport)],
             winner_outcome: &OutcomeReport,
         ) -> DispatchResult {
             let mut total_incentives = BalanceOf::<T>::from(0u8);
             let mut total_winners = BalanceOf::<T>::from(0u8);
 
-            for (jai, _, _, (_, outcome)) in
-                Self::non_tardy_jurors_with_votes(requested_jurors, votes)
-            {
-                if outcome == winner_outcome {
+            for (jai, outcome) in valid_winners_and_losers {
+                if outcome == &winner_outcome {
                     total_winners = total_winners.saturating_add(BalanceOf::<T>::from(1u8));
                 } else {
                     let all_reserved = CurrencyOf::<T>::reserved_balance_named(&RESERVE_ID, jai);
```

Comment:
In regards to line 360: It is not part of the current changes, but it would be nice to have constants like slash percentages as pallet parameters (50% is currently hard-coded).

Reply:
Created https://github.com/zeitgeistpm/zeitgeist/issues/309 to tackle this matter. Personally, runtime parameters are useful for different relay chains (Kusama and/or Polkadot) but third-party usage is questionable due to the specific and particular nature of this pallet.